pkg/kubelet/kubelet_pods.go

```diff
@@ -687,11 +687,7 @@ func (kl *Kubelet) podIsTerminated(pod *v1.Pod) bool {
 		// restarted.
 		status = pod.Status
 	}
-	if status.Phase == v1.PodFailed || status.Phase == v1.PodSucceeded {
-		return true
-	}
-
-	return false
+	return status.Phase == v1.PodFailed || status.Phase == v1.PodSucceeded || (pod.DeletionTimestamp != nil && notRunning(status.ContainerStatuses))
```

Comment:
This shouldn't have any effects during pod startup, as this change only makes a difference when the deletion timestamp is set.  I agree that we should do a more thorough review of what effects this could have, though.  In general, I think this is a better definition of podIsTerminated, as a pod that has been deleted, and has no containers running is definitely in a terminal state.